dali/operators/reader/webdataset_reader_op.cc

```diff
@@ -0,0 +1,94 @@
+      std::memcpy(ws.OutputRef<CPUBackend>(output_idx)[data_idx].raw_mutable_data(),
+                  sample[output_idx].raw_data(), sample[output_idx].nbytes());
+    }
+  }
+}
+
+DALI_SCHEMA(readers__Webdataset)
+    .DocStr(
+        R"code(A reader for the webdataset format.
+
+The webdataset format is a dataset meant to improve caching of the data between the samples read.
+The data itself is contained within one or several tar archives, each of which is further split into
+samples, each of which contains one or several components that correspond to the actual file
+contained within the archive. The components that correspond to specific sample are aggregated by
+the part of the filepath that does not correspond to the extension (for the specifics about the
+extensions please read the description of the ``ext`` parameter below). Note that samples with
+their filename starting with a dot will not be loaded, as well as entries that are not a file.
+
+In addition to the tar archive with data, each archive should come with a corresponding index file.
+The format of the index file is as follows:
+<offset behind the contents of the last archive entry> <number_of_samples>
+<sample1_start_offset> <component1_ext> <component1_size> <component2_ext> <component2_size> ...
+<sample2_start_offset> <component1_ext> <component1_size> <component2_ext> <component2_size> ...
+...)code")
+    .NumInput(0)
+    .OutputFn([](const OpSpec& spec) {
+      return spec.HasArgument("ext") ? spec.GetRepeatedArgument<std::string>("ext").size() : 0;
+    })
+    .AddArg("uris", R"code(The list of (one or more) paths to the webdataset archives.
+Has to be the same length as the ``configs`` argument.)code",
+            DALI_STRING_VEC)
+    .AddArg("configs",
+            R"code(The list of the index files corresponding to the respective webdataset archives.
+Has to be the same length as the ``uris`` argument.)code",
+            DALI_STRING_VEC)
+    .AddArg("ext", R"code(The extension sets for each of the outputs produced.
+The number of extension sets determines the number of outputs of the reader.
+The extensions of the components are counted as the text after the first dot in the name of the file 
+(excluding the samples starting with a dot). The different extension options should be separated
+with a semicolon (';') and may contain dots.
+
+Example: "left.png;right.jpg")code",
+            DALI_STRING_VEC)
+    .AddOptionalArg(
+        "missing_component_behavior",
+        R"code(Specifies what to do in case there is not any file corresponding to a certain output.
+Three behaviors are possible (case-insensitive):
+  - "empty" (default) - in that case the output that was not set will just contain an empty tensor
+  - "skip" - in that case the entire sample will just be skipped (no penalty to performance except
+             for reduced caching of the archive)
+  - "error" - in that case an exception will be raised)code",
```

Comment:
```suggestion
  - "error" - in that case an exception will be raised and the execution stops)code",
```

Reply:
If it happens in the constructor, it doesn't really start, so saying that it stops is misleading (sounds like late failure).

Reply:
I think it happens in runtime.